src/macros.rs

```diff
@@ -37,34 +39,65 @@ macro_rules! redis_command {
             )
         } == $crate::raw::Status::Err as c_int
         {
+            $crate::raw::redis_log(
+                $ctx,
+                &format!("Error: failed to create command {}", $command_name),
+            );
             return $crate::raw::Status::Err as c_int;
         }
 
-        if $acl_categories != "" {
-            let acl_categories = CString::new($acl_categories).unwrap();
+        let command =
+            unsafe { $crate::raw::RedisModule_GetCommand.unwrap()($ctx, name.as_ptr()) };
+        if command.is_null() {
+            $crate::raw::redis_log(
+                $ctx,
+                &format!("Error: failed to get command {}", $command_name),
+            );
+            return $crate::raw::Status::Err as c_int;
+        }
 
-            let command =
-                unsafe { $crate::raw::RedisModule_GetCommand.unwrap()($ctx, name.as_ptr()) };
-            if command.is_null() {
-                return $crate::raw::Status::Err as c_int;
-            }
+        if let Some(RM_SetCommandACLCategories) = $crate::raw::RedisModule_SetCommandACLCategories {
+            let mut acl_categories = CString::new("").unwrap();
+            $(
+                if $mandatory_acl_categories != "" && $optional_acl_categories != "" {
+                    acl_categories = CString::new(format!("{} {}", $mandatory_acl_categories, $optional_acl_categories)).unwrap();
+                } else if $optional_acl_categories != "" {
+                    acl_categories = CString::new($optional_acl_categories).unwrap();
+                }
```

Comment:
I don't like them being strings; I'd prefer having those as a type-safe `CommandAclCategory` enum, which would allow it to convert itself to a string, and a vector of those would be able to be joined with spaces and provide the same result. Just from the user perspective, the constant strings isn't the way to go, especially the number of times we repeat them.

Reply:
the point is to allow the user to provide any arbitrary categories they like. we cannot provide an exhaustive enum, because the set of categories is infinite